src/controllers/buses.controller.ts

```diff
@@ -99,3 +100,29 @@ export const deleteBusById = async (req: Request, res: Response)=> {
         data: {count: deleteCount}
     })
 }
+
+export const linkBusToRoute = async (req: Request, res: Response) => {
```

Comment:
I would suggest you wrap the code in this function in try-catch to display proper messages when an error occurs (like duplicate links).